Salesforce doc update (#475)
rudderlabs/rudderstack-docs: destinations/crm/salesforce.mdx

</div>

### Updating custom fields in Salesforce

<div class="infoBlock">

If you wish to update custom fields in Salesforce using RudderStack, make sure that you first create those lead fields in Salesforce before sending the data through RudderStack. For more information, refer to the <a href="https://help.salesforce.com/articleView?id=sf.adding_fields.htm&type=5" target="_blank">Salesforce Help</a> guide.
</div>

As the [Salesforce Leads API](https://developer.salesforce.com/docs/atlas.en-us.api.meta/api/sforce_api_objects_lead.htm) requires`lastName` and `company` to be present, absence of either of these fields will result in RudderStack automatically appending the `'n/a'` to both the fields - even if they have been specified in some previous request.

For example, if you want to collect a custom trait in RudderStack named `newProp`, create a field label named `newProp`. This will generate an API name as `newProp__c`. RudderStack automatically appends the `__c` to any custom trait.

<div class="warningBlock">

> <strong>Make sure you are consistent with your casing</strong>. If the custom fields are created in the camel case, then make sure sure that you send the traits to RudderStack in the camel case. If you're creating the custom fields in the snake case, make sure you send the traits in the same format.

Reviewer: ```suggestion
<strong>Make sure the casing is consistent </strong>. For example, if the custom fields are created in the camel case, then the traits must be sent to RudderStack in the camel case only.
```

</div>

## Updating Salesforce objects

You can create or update any [Salesforce object](https://developer.salesforce.com/docs/atlas.en-us.object_reference.meta/object_reference/sforce_api_objects_list.htm) using the `identify` event. To specify the object type, follow the schema below.

> RudderStack will look for the key `externalId` under `context` and determine the Salesforce object type by removing the part `Salesforce-` from the field `type`. It will then make a `PATCH` request if there is an `id` present in the request to update the record. Otherwise, a new record will be created.

Reviewer: ```suggestion
RudderStack looks for the key `externalId` under `context` and determines the Salesforce object type by removing the part `Salesforce-` from the field `type`. Further, it makes a `PATCH` request if there is an `id` present in the request to update the record. Otherwise, a new record is created.
```
